chaoscenter/graphql/server/pkg/probe/utils/utils.go

```diff
@@ -697,7 +697,7 @@ func ParseProbesFromManifestForRuns(wfType *dbChaosExperiment.ChaosExperimentTyp
 }
 
 // GenerateExperimentManifestWithProbes - uses GenerateProbeManifest to get and store the respective probe attribute into Raw Data template for Non Cron Workflow
-func GenerateExperimentManifestWithProbes(manifest string, projectID string) (argoTypes.Workflow, error) {
+func GenerateExperimentManifestWithProbes(manifest string, projectID string, probeOperator *dbSchemaProbe.Operator) (argoTypes.Workflow, error) {
```

Comment:
Rather than passing the operator as a parameter, better to add this to the Probe Service and then use it like so

```go
func (p *probe) GenerateExperimentManifestWithProbes ... {
    p.probeOperator.GetProbeByName
}
```

Otherwise every instance of this handler would require you to declare and pass on this probeOperator like this
`something := GenerateExperimentManifestWithProbes(..previous params, interface.probeOperator)`

The better approach would be to use this handler via the service
`something := c.probeService.GenerateExperimentManifestWithProbes(..previous params)`

Its easier to test in the long run, the test cases would actually be used as checkpoints with this approach. Which also means that the respective initializers would need to change. For example for the above

```go
type experimentRunOps struct {
	...previous values
	probeService  probeService.Service
}

func NewExperimentRunOps(...previous values, probeService probeService.Service) ExperimentRunOps {
	return &experimentRunOps{
		...previous values,
		probeService:  probeService,
	}
}
```

Reply:
I’ve made the changes as suggested. Thank you for the feedback.